realm/realm-library/src/main/java/io/realm/internal/OsCollectionChangeSet.java

```diff
@@ -45,12 +45,19 @@
 
     private static long finalizerPtr = nativeGetFinalizerPtr();
     private final long nativePtr;
+    private final boolean firstAsyncCallback;
 
-    public OsCollectionChangeSet(long nativePtr) {
+    public OsCollectionChangeSet(long nativePtr, boolean firstAsyncCallback) {
         this.nativePtr = nativePtr;
+        this.firstAsyncCallback = firstAsyncCallback;
         NativeContext.dummyContext.addReference(this);
     }
 
+    @Override
+    public State getState() {
+        throw new UnsupportedOperationException("This method should be overridden in a subclass");
```

Comment:
Why not make the class abstract instead, and not implement this method here?

Reply:
I agree this is not pretty. The reason for this is mostly because `OsX` sort of indicates a wrapper around an ObjectStore class. This also implies that we probably shouldn't add extra responsibility to it. Which is where the `StatefulCollectionChangeSet` comes from. These methods are only here because `OsCollectionChangeset` currently need to implement the public interface `OrderedCollectionChangeSet`, but these methods don't make sense here.

I should probably take a second look at this to see if it could be made cleaner.